### PR TITLE
chore: pylint fix

### PR DESCRIPTION
**Pull request by GavinZZ**
### Description of changes
Enabling more pyling rules. Putting more strict requirement on the code.

### Description of how you validated changes
Ran integ tests and all passed.
`2739 passed in 56.52s`

### CODE REVIEW

File: samtranslator/plugins/__init__.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def on_before_transform_resource(self, logical_id, resource_type, resource_prope
         # NoOp implementation
         pass
 
-    def on_before_transform_template(self, template):
+    def on_before_transform_template(self, template_dict):
```
Comment (hoffa) [inline samtranslator/plugins/__init__.py:59]: Why?

**Review by hoffa — APPROVED** [review]
Nice 👍

File: samtranslator/plugins/__init__.py (hunk 1 above)
Comment (GavinZZ) [reply samtranslator/plugins/__init__.py:59]: This is changed due to pyling rule W0237. 
```
Used when a method parameter has a different name than in the implemented interface or in an overridden method. 
```
https://pylint.pycqa.org/en/latest/user_guide/messages/warning/arguments-renamed.html

File: samtranslator/model/stepfunctions/generators.py [hunk 2]
```diff
@@ -262,7 +258,7 @@ def _generate_event_resources(self):
                     for name, resource in self.event_resources[logical_id].items():
                         kwargs[name] = resource
                 except (TypeError, AttributeError) as e:
-                    raise InvalidEventException(logical_id, "{}".format(e))
+                    raise InvalidEventException(logical_id, f"{e}")
```
Comment (aahung) [inline samtranslator/model/stepfunctions/generators.py:261]:
```suggestion
                    raise InvalidEventException(logical_id, str(e))
```

**Review by aahung — APPROVED** [review]

**Pull request merged**